MRG: Update docs

Eric89GXL opened the pull request:
Closes #276.
Closes #277.

~~WIP because building freezes on my machine at `Embedding documentation hyperlinks in examples ...` (40%) so I want to see what CircleCI gives, and I still want/need to address #277.~~

Eric89GXL commented:
Old docs:

http://sphinx-gallery.readthedocs.io/en/latest/advanced_configuration.html

This PR's version:

https://154-25860190-gh.circle-artifacts.com/0/home/ubuntu/sphinx-gallery/rtd_html/advanced_configuration.html

Ready for review/merge from my end.

Titan-C commented on README.rst:
```diff
@@ -103,9 +108,7 @@ structure you get from executing ``sphinx-quickstart``.
-``doc/conf.py`` file with:
-
-.. code-block:: python
+``doc/conf.py`` file with::
```
I'm just curious. Why do you make bash code blocks now explicit, make the python blocks now un-explicit?

Eric89GXL commented:
Python is the default, so specifying it wastes space (most modules I've
seen don't bother to specify when it's Python)

choldgraf commented on doc/advanced_configuration.rst:
```diff
@@ -1,42 +1,70 @@
+=============
+Configuration
+=============
+
```
I'd start with a short description of what is to come. Something like

```
Configuration and customization of sphinx-gallery is done with a dictionary specified in your `conf.py` file. A list of the possible keys and their corresponding values are explained in this section, and are explained in greater detail elsewhere in the documentation.
```

choldgraf replied:
(edited this slightly, just in case you're reading from your email ;-) )

choldgraf commented on doc/advanced_configuration.rst:
```diff
@@ -1,42 +1,70 @@
+
+And some things can be tweaked directly in CSS:
+
+- ``.sphx-glr-thumbcontainer`` (:ref:`setting_thumbnail_size`)
```
did we ever add the more descriptive classes for other objects (e.g. DIVs etc)?

Titan-C replied:
all our CSS is in sphinx_gallery/_static/gallery.css, and every user can always overload that with their own CSS. I left PR #210 forgotten, at some point we shall go back to it.

choldgraf replied:
ah yeah - I meant using more sphinx-gallery-specific classes in the DIVs...I'll comment over in #210!

choldgraf commented:
This looks almost ready to go for me - two quick comments but then I am +1 on merge

Eric89GXL commented:
@choldgraf can you see if you're happy with the latest header change?

The only remaining question in my mind is is this list of options complete? I saw at least `find_mayavi_figures` not documented, so I added that one.

I also saw over in #210 something about `multi-img` and it made me think there was some config var that allows using multiple thumbnail images (?) but I don't see it documented anywhere. From looking through the code with `gallery_conf` I don't see any other entries missing, so I guess not.

Eric89GXL commented:
Latest build:

https://157-25860190-gh.circle-artifacts.com/0/home/ubuntu/sphinx-gallery/rtd_html/advanced_configuration.html

choldgraf commented:
I think we should merge this, and open an issue about the more general challenge of "how do we ensure that all the functionality of SG (within the conf.py file) is tested and documented"...I think the repo's `conf.py` file should have an example of each possible config option, but I could see folks disagreeing

choldgraf commented:
I'll merge this tomorrow AM if nobody protests :-)

Eric89GXL commented:
Okay I've also added these to the `conf.py`:
```
# See warnings about bad links
nitpicky = True
# we intentionally link outside images
suppress_warnings = ['image.nonlocal_uri']
```
... because without the first, I didn't know that [the links I added](https://158-25860190-gh.circle-artifacts.com/0/home/ubuntu/sphinx-gallery/rtd_html/advanced_configuration.html#find-mayavi) to `:mod:matplotlib` and `:mod:mayavi` were no good (and needed to be `:mod:matplotlib.pyplot` and `:mod:mayavi.mlab`), and without the second, we get extra warnings that we don't need to see.

This should at least make visible other similar errors in the future.

choldgraf commented:
I'm not sure what the `nitpicky` change means...is that just a sphinx param to throw an error if it finds a warning?

Eric89GXL commented:
No, it just gives more warnings about bad links

Eric89GXL commented:
Ping @choldgraf this should be good to go

choldgraf commented:
🎉🎉 🎉 🎉
